src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -63,7 +63,17 @@ def is_dxlink(x):
     Returns whether *x* appears to be a DNAnexus link (is a dict with
     key ``"$dnanexus_link"``) with a referenced data object.
     '''
-    return isinstance(x, dict) and '$dnanexus_link' in x and (isinstance(x['$dnanexus_link'], basestring) or isinstance(x['$dnanexus_link'], dict) and 'id' in x['$dnanexus_link'])
+    if not isinstance(x, dict):
+        return False
+    if '$dnanexus_link' not in x:
+        return False
+    link = x['$dnanexus_link']
+    if isinstance(link, basestring):
+        return True
```

Comment:
@mckinsel, this doesn't break _guess_link_target_type, because that function already craps out with an exception when an unexpected link format is encountered. It will continue to throw an exception in the same manner for JBORs.